golang.go

```diff
@@ -130,27 +130,67 @@ type GolangConfigBuild struct {
 	// Remove all filesystem paths from the resulting executable
 	TrimPath bool
 
-	magnet *Magnet
+	paths  containerPathMapping
+	target *MagnetTarget
 }
 
-func (m *GolangConfigBuild) cacheDir() string {
-	path := filepath.Join(m.magnet.AbsCacheDir(), "go")
-	os.MkdirAll(path, 0755)
-	return path
+// BuildContainer describes a build container image
+type BuildContainer struct {
+	// Name identifies the container image
+	Name string
+	// GOPath optionally overrides the gopath inside the container.
+	//
+	// If set, the module path inside the container (ContainerPath)
+	// will be automatically deduced as `GOPath/src/ModulePath`.
+	// The value of the attribute will be written as `GOPATH` envar
+	// inside the container.
+	// Also sets `GO111MODULE=auto` envar if unspecified.
+	GOPath string
```

Comment:
This is what I came up with instead of inferring the GOPATH from host. I think container has little to do with the host environment and the bit requiring the configuration is whether to use GOPATH and where it's at inside the container.